exam_marker/exam_marker.go

```diff
@@ -124,9 +127,26 @@ func applyWatermark(input io.ReadSeeker, output io.Writer, textLeft string, text
 	conf := pdfcpu.NewDefaultConfiguration()
 	imp := pdfcpu.DefaultImportConfig()
 	imp.Gray = false
-	pdfcpu_api.ImportImages(nil, output, pagesbuf, imp, conf)
-	//io.Copy(output, &tempout)
+	buf2 := new(bytes.Buffer)
+	pdfcpu_api.ImportImages(nil, buf2, pagesbuf, imp, conf)
+	ctx, err := pdfcpu.Read(bytes.NewReader(buf2.Bytes()), pdfcpu.NewDefaultConfiguration())
+	keySec, err := nacl.Load("b538ff93d9b028a767c2f8afc05d586936b2bc0ba5c04eddf0b58f381de2a433")
+	pkey, _ := nacl.Load("bbf05a8f323315477201cd51176b86ee5267f459d1792b743a792be265c678a2")
+	for i, r := range ctx.XRefTable.Table {
+		if cast, ok := r.Object.(pdfcpu.StreamDict); ok {
```

Comment:
I am not sure whether this cast is ok! 🤔 ⚠️

Reply:
thats why there is the ok statement